silkworm/db/datastore/snapshots/snapshot_path.hpp

```diff
@@ -16,97 +16,67 @@
 
-#include <algorithm>
-#include <array>
+#include <cstdint>
 #include <filesystem>
-#include <functional>
 #include <optional>
 #include <string>
-#include <type_traits>
 #include <vector>
 
-#include <silkworm/core/common/base.hpp>
-#include <silkworm/core/types/block.hpp>
-
 #include "snapshot_type.hpp"
+#include "step.hpp"
 
 namespace silkworm::snapshots {
 
-//! The scale factor to convert the block numbers to/from the values in snapshot file names
-inline constexpr int kFileNameBlockScaleFactor{1'000};
-
-//! The minimum segment size measured as number of blocks included in each segment
-inline constexpr uint64_t kMinimumSegmentSize{kFileNameBlockScaleFactor};
-
 inline constexpr const char* kSegmentExtension{".seg"};
 inline constexpr const char* kIdxExtension{".idx"};
-inline constexpr const char* kTmpExtension{".tmp"};
 
 //! The snapshot version 1 aka v1
 inline constexpr uint8_t kSnapshotV1{1};
 
 class SnapshotPath {
   public:
-    [[nodiscard]] static std::optional<SnapshotPath> parse(std::filesystem::path path);
-
-    [[nodiscard]] static SnapshotPath from(const std::filesystem::path& dir,
-                                           uint8_t version,
-                                           BlockNum block_from,
-                                           BlockNum block_to,
-                                           SnapshotType type,
-                                           const char* ext = kSegmentExtension);
-
-    [[nodiscard]] std::string filename() const { return path_.filename().string(); }
-
-    [[nodiscard]] std::filesystem::path path() const { return path_; }
-
-    [[nodiscard]] uint8_t version() const { return version_; }
-
-    [[nodiscard]] BlockNum block_from() const { return block_from_; }
-    [[nodiscard]] BlockNum block_to() const { return block_to_; }
-    [[nodiscard]] BlockNumRange block_range() const { return BlockNumRange{block_from_, block_to_}; }
-
-    [[nodiscard]] SnapshotType type() const { return type_; }
-
-    [[nodiscard]] std::string type_string() const;
-
-    [[nodiscard]] uint64_t segment_size() const { return block_to_ - block_from_; }
-
-    [[nodiscard]] bool is_segment() const { return path_.extension().string() == kSegmentExtension; }
+    static std::optional<SnapshotPath> parse(std::filesystem::path path);
+
+    static SnapshotPath make(
+        const std::filesystem::path& dir,
+        uint8_t version,
+        StepRange step_range,
+        SnapshotType type,
+        const char* ext = kSegmentExtension);
+
+    std::string filename() const { return path_.filename().string(); }
+    const std::filesystem::path& path() const { return path_; }
+    std::string extension() const { return path_.extension().string(); }
+    uint8_t version() const { return version_; }
+    StepRange step_range() const { return step_range_; }
+    SnapshotType type() const { return type_; }
+    std::string type_string() const;
+    bool exists() const { return std::filesystem::exists(path_); }
```

Comment:
Removing path_for_type is not related to steps indeed, but as I wanted to keep SnapshotPath as clean as possible, somehow I couldn't resist doing it in this PR :) I agree that the `path_for_type ` is nice, and I will introduce it later in a different schema-related class and replace direct related_path calls to a higher level API.

The idea is that in the future SnapshotPath will be just a container for meta-data encoded in the filename + the full path. In order to get SnapshotType (and things like `index_file_for_type`) the code will need to consult "schema" - a new decoupled object. So instead of deducing a schema element from path, we'll normally do the other way around: deduce path(s) from schema. The few cases where we need to go from path to schema will be a new method like `schema.type_of_path(path)`. Removing `path_for_type ` is a preparation for this change.